crates/query-engine/translation/tests/snapshots/tests__select_where_album_id_equals_self_nested_object_relationship.snap

```diff
@@ -30,100 +30,116 @@ FROM
-                              ) AS "%6_RELATIONSHIP_Artist" ON ('true')
+                                              ("%7_Album"."ArtistId" = "%9_Artist"."ArtistId")
+                                          ) AS "%10_rows"
+                                      ) AS "%10_rows"
+                                  ) AS "%8_RELATIONSHIP_Artist"
+                              ) AS "%8_RELATIONSHIP_Artist" ON ('true')
                             WHERE
-                              ("%0_Track"."AlbumId" = "%5_Album"."AlbumId")
-                          ) AS "%10_rows"
-                      ) AS "%10_rows"
+                              ("%0_Track"."AlbumId" = "%7_Album"."AlbumId")
+                          ) AS "%12_rows"
+                      ) AS "%12_rows"
                   ) AS "%1_RELATIONSHIP_Album"
               ) AS "%1_RELATIONSHIP_Album" ON ('true')
-              INNER JOIN LATERAL (
-                SELECT
-                  *
-                FROM
-                  "public"."Album" AS "%2_BOOLEXP_Album"
-                WHERE
-                  (
-                    (
-                      "%2_BOOLEXP_Album"."Title" = cast($1 as "pg_catalog"."varchar")
-                    )
-                    AND (
-                      "%0_Track"."AlbumId" = "%2_BOOLEXP_Album"."AlbumId"
-                    )
-                  )
-              ) AS "%2_BOOLEXP_Album" ON ('true')
-              INNER JOIN LATERAL (
+            WHERE
+              EXISTS (
                 SELECT
-                  *
+                  1
                 FROM
-                  "public"."Album" AS "%3_BOOLEXP_Album"
-                WHERE
                   (
-                    (
-                      "%3_BOOLEXP_Album"."Title" = cast($2 as "pg_catalog"."varchar")
-                    )
-                    AND (
-                      "%0_Track"."AlbumId" = "%3_BOOLEXP_Album"."AlbumId"
-                    )
-                  )
-              ) AS "%3_BOOLEXP_Album" ON ('true')
-              INNER JOIN LATERAL (
-                SELECT
-                  *
-                FROM
-                  "public"."Artist" AS "%4_BOOLEXP_Artist"
+                    SELECT
+                      "%2_BOOLEXP_Album".*
+                    FROM
+                      (
+                        SELECT
+                          *
+                        FROM
+                          "public"."Album" AS "%2_BOOLEXP_Album"
+                        WHERE
+                          (
+                            (
+                              "%2_BOOLEXP_Album"."Title" = cast($1 as "pg_catalog"."varchar")
+                            )
+                            AND (
+                              "%0_Track"."AlbumId" = "%2_BOOLEXP_Album"."AlbumId"
+                            )
+                          )
+                      ) AS "%2_BOOLEXP_Album"
+                  ) AS "%3_BOOLEXP_Album" FULL
+                  OUTER JOIN LATERAL (
```

Comment:
Say we want to write a query with a predicate like:

```graphql
where: {
  _or: [
    {Albums: {Title: {_ilike: "%e%"}}},
    {Managers: {Name: {_ilike: "A%"}}}
  ]
}
```

The SQL predicate will be `Album.Title ilike '%e%' OR Manager.Name ilike 'A%'`.

To use this predicate, we need to extract the columns from the relationship tables.

We want to extract the titles of albums, and also extract the names of managers.
We have a few cases for the artist:

1. The artist has managers and albums
2. _ has only managers
3. _ has only albums
4. _ has neither

As a join table:

| has      | managers.name | albums.title |
| -------- | ------------- | ------------ |
| both     | value       | value      |
| managers | value      | null         |
| albums   | null          | value      |
| neither  | null          | null         |

1. inner join will only fetch row (1)
2. left join will only fetch the rows (1,2)
3. full join will fetch rows (1,2,3)

In a query with an OR expression, it is important to handle all three cases, because even if we don't have tracks, we still want to return the row if the artist has a manager named `Alfred`.

Reply:
I would love to see this note in the code comments, or at least a link to this PR, as it's really well explained.

Reply:
Added a link to this comment in the code comments :)